src/Nethermind/Nethermind.Trie/HexPrefix.cs

```diff
@@ -39,23 +42,45 @@ public static byte[] ToBytes(byte[] path, bool isLeaf)
 
         public static (byte[] key, bool isLeaf) FromBytes(ReadOnlySpan<byte> bytes)
         {
-            bool isLeaf = bytes[0] >= 32;
             bool isEven = (bytes[0] & 16) == 0;
             int nibblesCount = bytes.Length * 2 - (isEven ? 2 : 1);
             byte[] path = new byte[nibblesCount];
-            for (int i = 0; i < nibblesCount; i++)
+            Span<byte> span = new(path);
+            if (!isEven)
+            {
+                span[0] = (byte)(bytes[0] & 0xF);
+                span = span.Slice(1);
+            }
+            bool isLeaf = bytes[0] >= 32;
+            bytes = bytes.Slice(1);
+
+            Span<ushort> nibbles = MemoryMarshal.CreateSpan(
+                ref Unsafe.As<byte, ushort>(ref MemoryMarshal.GetReference(span)),
+                span.Length / 2);
+
+            Debug.Assert(nibbles.Length == bytes.Length);
+
+            ref byte byteRef = ref MemoryMarshal.GetReference(bytes);
+            ref ushort lookup16 = ref MemoryMarshal.GetArrayDataReference(Lookup16);
+            for (int i = 0; i < nibbles.Length; i++)
```

Comment:
this looks SIMD'able?

Reply:
I think it needs a deeper look; like do we need to expand to nibbles and convert back at all? 🤔